Engine/DataFeeds/UniverseSelection.cs

```diff
@@ -238,10 +237,14 @@ public SecurityChanges  ApplyUniverseSelection(Universe universe, DateTime dateT
             {
                 // create the new security, the algorithm thread will add this at the appropriate time
                 Security security;
-                if (!pendingAdditions.TryGetValue(symbol, out security) && !_algorithm.Securities.TryGetValue(symbol, out security))
+                if (!pendingAdditions.TryGetValue(symbol, out security))
                 {
-                    security = universe.CreateSecurity(symbol, _algorithm, _marketHoursDatabase, _symbolPropertiesDatabase);
-                    pendingAdditions.Add(symbol, security);
+                    security = universe.GetSecurity(symbol, _algorithm);
+                    if (security == null)
+                    {
+                        security = universe.CreateSecurity(symbol, _algorithm, _marketHoursDatabase, _symbolPropertiesDatabase);
+                        pendingAdditions.Add(symbol, security);
```

Comment:
RE: line 270
> additions.Add(security);

Should we be sending `SecuritiesChanged` events for the benchmark? Maybe a check for internals? I'm guessing in general we should be sending these events for any internal subscriptions (maybe a separate PR issue) -- but may cause issues in `OnSecuritiesChanged` if someone does something with the security and it's really the benchmark security and not the universe generated security.

Reply:
My initial thought was that internal/benchmark security additions should never fire `OnSecurityChanged`.
When `ApplyUniverseSelection` is called for the benchmark universe, the `SecurityChanges` return value is not currently used.